SaveContext.py

```diff
@@ -1,4 +1,25 @@
 from itertools import chain
+from enum import IntEnum
+
+class Scenes(IntEnum):
+    DEKU_TREE = 0x00
+    DODONGOS_CAVERN = 0x01
+    KING_DODONGO_LOBBY = 0x12
+    JABU_JABU = 0x02
+    FOREST_TEMPLE = 0x03
+    FIRE_TEMPLE = 0x04
+    WATER_TEMPLE = 0x05
+    SPIRIT_TEMPLE = 0x06
+    SHADOW_TEMPLE = 0x07
+
+class FlagType(IntEnum):
+    CHEST = 0x00
+    SWITCH = 0x04
+    CLEAR = 0x08
+    COLLECT = 0x0C
+    # 0x10 unused
+    VISITED_ROOM = 0x14
+    VISITED_FLOOR = 0x18
```

Comment:
Pretty minor, but it might make more sense to have a enum of types 0 to 6 and do type * 4 in the `write_permanent_flag` function below, since the * 4 technically comes from the 4 bytes used per type

Reply:
Made the change, makes sense since I already did this for the Scenes enum.